cmd/clusterctl/client/tree/discovery_test.go

```diff
@@ -286,7 +748,8 @@ func Test_Discovery(t *testing.T) {
 
 			for parent, wantChildren := range tt.wantTree {
 				gotChildren := tree.GetObjectsByParent(types.UID(parent))
-				g.Expect(wantChildren).To(HaveLen(len(gotChildren)), "%q doesn't have the expected number of children nodes", parent)
+				g.Expect(gotChildren).To(HaveLen(len(wantChildren)), "%q doesn't have the expected number of children nodes", parent)
```

Comment:
@fabriziopandini I'm wondering if the `gotChildren` and `wantChildren` here should be reversed so it would look like `g.Expect(gotChildren).To(HaveLen(len(wantChildren)))`. If I understand correctly typically `g.Expect()` takes the result from the function we are testing right?